src/errors.ts

```diff
@@ -7,13 +7,28 @@ export class MalanError extends Error {
   }
 }
 
+type ErrorDetails = {
+  string: [string]
+}
+
 type Response = {
   response: {
     body: {
-      errors?: {
-        detail: string;
-        message: string;
-      };
+      ok: boolean;
+      code: 400 | 401 | 403 | 404 | 422 | 423 | 429 | 461 | 462 | 500;
+      message: string;
+      errors?: ErrorDetails;
+      detail:
+        | "Bad Request"
+        | "Unauthorized"
+        | "Forbidden"
+        | "Not Found"
+        | "Unprocessable Entity"
+        | "Locked"
+        | "Too Many Requests"
+        | "Terms of Service Required"
+        | "Privacy Policy Required"
+        | "Internal Server Error";
```

Comment:
Union types can provide a lot of utility in cases like this because they allow Typescript to perform type narrowing:

```
type ErrorResponse = { ok: false, ... };
type SuccessResponse: { ok: true, ... };
type Response = {
  response: {
    body: ErrorResponse | SuccessResponse
  }
}
```

Then a simple conditional is enough to tell Typescript what other assumptions are safe:

```
if (response.body.ok === false) {
  // Typescript knows that response.body also has a code, detail, message, etc.
}
```

It's possible to take this even further with a type for each unique [code, detail] tuple. That could yield diminishing returns. Discriminating the error and success types is a big win.